config/src/network_id.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct NetworkInfo {
+    name: String,
+}
+
+/// A representation of the network being used in communication.
+/// There should only be one of each NetworkId used for a single node, and handshakes should verify
+/// that the NetworkId being used is the same during a handshake, to effectively ensure communication
+/// is restricted to a network.  Network should be checked that it is not the `DEFAULT_NETWORK`
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub enum NetworkId {
+    Validator,
+    Public,
+    Private(NetworkInfo),
+}
+
+const INVALID: &str = "Invalid";
+const VFN: &str = "VFN";
+
+impl Default for NetworkId {
+    fn default() -> NetworkId {
+        NetworkId::private_network(INVALID.to_string())
+    }
+}
+
+impl NetworkId {
+    /// Tell us whether the `NetworkId` was configured correctly.
+    /// Private networks must have a name, otherwise we can't differentiate.
+    pub fn is_valid(&self) -> bool {
+        match self {
+            NetworkId::Validator => true,
+            NetworkId::Public => true,
+            NetworkId::Private(network_info) => network_info.name == INVALID,
+        }
+    }
```

Comment:
they should all be valid imo, not sure what we get by having this?

Reply:
Same as above.  I'll check if the Default value was because of the struct or an enum.

Reply:
The purpose of this was that there is a `Default` for `NetworkConfig` when deserializing.  However, I wanted to be explicit that if `NetworkConfig` fills in the default value, that we should not trust it because it loaded a default value from the `NetworkConfig`.

I'm not sure having a defaulting configuration is really what we want on a `NetworkConfig`